packages/destination-actions/src/destinations/salesforce/sf-operations.ts

```
@@ -161,18 +161,31 @@ export default class Salesforce {
   }
 
   // Salesforce SOQL spec requires any single quotes to be escaped.
-  private escapeQuotes = (value: string) => value.replace(/'/g, "\\'")
+  private escapeQuotes = (value: string) => value.toString().replace(/'/g, "\\'")
 
   // Salesforce field names should have only characters in {a-z, A-Z, 0-9, _}.
   private removeInvalidChars = (value: string) => value.replace(/[^a-zA-Z0-9_]/g, '')
 
+  // Preformats trait values based on datatypes for correct SOQL syntax
+  private typecast = (value: any) => {
+    switch (typeof value) {
+      case 'boolean':
+        return value
+      case 'number':
+        return value
+      default:
```

I wonder if we should throw an error here when users try to lookup by object and array types? I think those sort of lookups would fail with 404 anyways and so it would be better to have a descriptive error message: (400: "Invalid lookup type: Array/Object")

---

Done. I have added a test case for that as well.